packages/relay/src/lib/eth.ts

```diff
@@ -177,29 +177,58 @@ export class EthImpl implements Eth {
     this.logger.trace(`${requestIdPrefix} feeHistory(blockCount=${blockCount}, newestBlock=${newestBlock}, rewardPercentiles=${rewardPercentiles})`);
 
     try {
-      const latestBlockNumber = await this.translateBlockTag(EthImpl.blockLatest, requestId);
-      const newestBlockNumber = (newestBlock == EthImpl.blockLatest || newestBlock == EthImpl.blockPending)
-        ? latestBlockNumber
-        : await this.translateBlockTag(newestBlock, requestId);
+      const resolvedBlockCount = blockCount > maxResults ? maxResults : blockCount;
+      if (resolvedBlockCount <= 0) {
+        return EthImpl.feeHistoryZeroBlockCountResponse;
+      }
 
-      if (newestBlockNumber > latestBlockNumber) {
-        return predefined.REQUEST_BEYOND_HEAD_BLOCK(newestBlockNumber, latestBlockNumber);
+      const cacheKey = `${constants.CACHE_KEY.FEE_HISTORY}_${blockCount}_${newestBlock}_${rewardPercentiles?.join('')}`;
+      let feeHistory: object | undefined = this.cache.get(cacheKey);
+      if(feeHistory) {
+        this.logger.trace(`${requestIdPrefix} returning cached value ${cacheKey}:${JSON.stringify(feeHistory)}`);
+        return feeHistory;
       }
 
-      blockCount = blockCount > maxResults ? maxResults : blockCount;
+      // check for latest network fees. Ensure cache is populated
+      const lastFeeChangeBlockCacheKey = `${constants.CACHE_KEY.FEE_HISTORY}_${constants.CACHE_KEY.ETH_BLOCK_NUMBER}}`;
+      let lastFeeChangeBlock = this.cache.get(lastFeeChangeBlockCacheKey);
+      if (!lastFeeChangeBlock) {
+        lastFeeChangeBlock = await this.getAndCacheLatestNetworkFeeChangeBlock(requestId);
```

Comment:
I had another idea here.

With the cache block of the last fee change we also have a timestamp to help inform what `latest` string refers to.
If we cache one more block, the block maxResults after the blcok that set the fee, then we can do the following trick.
When you encounter `latest` check the current timestamp, if it's above the (feeCHangeBlock + maxResults).timestamp then again there's no need to call the Mirror Node.
We just create the repeated History.

In this way blocks API need only be called once every hour at most based on activity. After that only if a numeric tag is provided for `newestBlock` whose value - blockCount is less that the `feeCHangeBlock` would we need to call blocks.